config/config.go

```diff
@@ -76,6 +78,29 @@ func NewConfig() *Configuration {
 	return c
 }
 
+// NewFromValues returns a new *config.Configuration instance as if it had
+// been read from the .gitconfig specified by "gitconfig" parameter.
+//
+// NOTE: this method should only be called during testing.
+func NewFromValues(gitconfig map[string]string) *Configuration {
+	config := &Configuration{
```

Comment:
Originally this `func` was in the `config_test.go` file, but I needed to use it in the [`commands_test.go`](https://github.com/github/git-lfs/pull/1257/files#diff-b65ac838606a4ab61fdb7724d458246dR11). This is sort of smelly to me, but I'm not quite sure what to do about it.